app/controllers/api/v2/concerns/api_authorizer.rb

```diff
@@ -19,7 +19,9 @@ def hosts_permission
 
   def resource_scope(_options = {})
     @resource_scope ||= begin
-      scope = PreupgradeReport.joins(:host).merge(Host.authorized(:view_hosts, Host))
+      scope = PreupgradeReport.includes(:preupgrade_report_entries)
```

Comment:
We shouldn't include entries here, because for `Api::V2::PreupgradeReportsController` actions `:index` and `:job_invocation` we don't need entries at all.